### ISSUE TITLE
Firejail Incompatible with Ubuntu 21.10

### ISSUE THREAD

**Issue by bobsmitherton** [netblue30/firejail#4625]
**Summary:** The firejail packet in the packet sources of Ubuntu 21.10 misses the suid bit.
**Fix:** `sudo chmod u+s /usr/bin/firejail`
If you have a launchpad account: https://bugs.launchpad.net/ubuntu/+source/firejail/+bug/1948480/comments/4

---

### Description

Upon upgrading to Ubuntu 21.10, Firejail no longer sandboxes Firefox as designed.
The Firefox browser is no longer restricted to just accessing the /Downloads directory.
In addition, LibreOffice Calc, Impress, and Writer won't start at all.
Hopefully, this security issue can be resolved quickly because I am browsing the Internet without the benefit of a sandbox at this time.

### Steps to Reproduce

**Part 1:** Launching FireTools results in:

>     Cannot run Firejail sandbox, you may not have the correct permissions to access this program.

In the terminal, I get this:

```console
$ firejail --list
Error PR_CAPBSET_DROP: caps.c:323 caps_drop_all: Operation not permitted
Error: failed to run /usr/bin/firemon
```

**Part 2:** Once Firejail is removed using `sudo apt-get purge --auto-remove firejail`, the LibreOffice programs start normally.
The LibreOffice programs **can** work if Firejail is installed **but** `sudo firecfg` **has not been run yet** (confirmed using `sudo firecfg --list` which shows no entries).

**Part 3:** Running `firejail firefox` kicks back this message:

>     Error mkdir: util.c:1138 create_empty_dir_as_root: Permission denied.

### Expected behavior

Firefox browser should be restricted to just accessing the /Downloads directory after running `sudo firecfg`.
LibreOffice programs should open.

### Actual behavior

I was able to successfully upload test files within Firefox from both the /Pictures and /Music directories thus confirming the sandbox is broken.
LibreOffice programs never start.

### Additional context

Furthermore, I tried the PPA located at: https://launchpad.net/~deki/+archive/ubuntu/firejail and that did not solve the issue.

### Environment

* Ubuntu: 21.10
* Firefox: 93.0
* Kernel: Linux 5.13.0-19-generic x86_64
* Firejail: 0.9.66-1~0ubuntu21.10.0
* LibreOffice: 7.2.2.2

---

Edit by @kmk3: Formatting.

**Comment by kmk3** [comment]
@bobsmitherton commented [on Oct 19](https://github.com/netblue30/firejail/issues/4625#issue-1030526477):

> ### Description
> 
> Upon upgrading to Ubuntu 21.10, Firejail no longer sandboxes Firefox as designed.
> The Firefox browser is no longer restricted to just accessing the /Downloads directory.
> In addition, LibreOffice Calc, Impress, and Writer won't start at all.
> Hopefully, this security issue can be resolved quickly because I am browsing the Internet without the benefit of a sandbox at this time.
>
> ### Steps to Reproduce
> 
> **Part 1:** Launching FireTools results in:
> 
> > ```
> > Cannot run Firejail sandbox, you may not have the correct permissions to access this program.
> > ```
> 
> In the terminal, I get this:
> 
> ```
> $ firejail --list
> Error PR_CAPBSET_DROP: caps.c:323 caps_drop_all: Operation not permitted
> Error: failed to run /usr/bin/firemon
> ```
> 
> **Part 2:** Once Firejail is removed using `sudo apt-get purge --auto-remove firejail`, the LibreOffice programs start normally. The LibreOffice programs **can** work if Firejail is installed **but** `sudo firecfg` **has not been run yet** (confirmed using `sudo firecfg --list` which shows no entries).
> 
> **Part 3:** Running `firejail firefox` kicks back this message:
> 
> > ```
> > Error mkdir: util.c:1138 create_empty_dir_as_root: Permission denied.
> > ```

Not sure if this is related or not, but Ubuntu 21.10 comes with Firefox (among
other programs) installed as a snap rather than a normal package (see #4554).
If the default "app store" is used, it also installs certain things as snaps.
And firejail does not work with snaps.

Can you confirm that the programs being executed aren't the snap versions?

**Comment by rusty-snake** [comment]
What did I said? Only 5 days. Fuck yeah, Canonical this time you really shot it down. How long do you actually want to use snaps before you use flatpaks? As long as you used upstart? Or Unity? Or MIR? Did any "innovation" from ubuntu ever make it? Or was it always Red Hat?

**Comment by reinerh** [comment]
The fixed version is now released in Ubuntu 21.10.
